chore: Bump `posthog-node` to v5+

rafaeelaudibert opened the pull request:
SDK Doctor was complaining we're in an outdated version of `posthog-node`, and we definitely are! Let's upgrade.

Breaking changes don't affect us (Node v20+) so this should be very safe. There's a bunch of error tracking upgrades as well so we might be able to start tracking these now - not changing anything on this PR

rafaeelaudibert commented:
@daibhin maybe we could remove this line now?

https://github.com/PostHog/posthog/blob/master/plugin-server/src/utils/posthog.ts#L8-L13

rafaeelaudibert committed:
fix: Remove unused `severity` argument

daibhin commented:
> maybe we could remove this line now?

We have added client-side rate limiting since so the "noise" should be a lot less. Will leave it up to the Ingestion folks to decide if they want to enable it though (not sure who else would need to debug plugin server exceptions)

rafaeelaudibert committed:
chore: Bump `posthog-node` to v5+

SDK Doctor was complaining we're in an outdated version of `posthog-node`, and we definitely are! Let's upgrade.

Breaking changes don't affect us (Node v20+) so this should be very safe. There's a bunch of error tracking upgrades as well so we might be able to start tracking these now - not changing anything on this PR

eli-r-ph reviewed (approved):
Not the most TypeScript-native person on the team 😆 but after reviewing potentially breaking changes vs. the uses ingestion is putting `posthog-node` to atm, this upgrade appears safe to me 👍
